### PR TITLE
Vdf security parameters update

### PR DESCRIPTION
**Pull request by 0o-de-lally**
Implements: https://github.com/OLSF/libra/issues/777

- [x] change Move TowerState APIs to accept difficulty and security params.
- [x] change Tower and TXS apps to send difficulty and security params.
- [x] benchmarks for different VDF parameters.
- [x] Tower when parsing blocks if does not find difficulty or security param defaults to "legacy" params of 5M iterations and 2048bit proofs.

### CODE REVIEW

File: language/diem-framework/modules/0L/TowerState.move [hunk 1]
```diff
@@ -21,6 +21,8 @@ module TowerState {
     use 0x1::VDF;
     use 0x1::Vector;
 
+    const VDF_SECURITY_PARAM: u64 = 2048;
```
Comment (zmanian) [inline language/diem-framework/modules/0L/TowerState.move:24]: Why is this 2048? I thought the point was to move 256?

Comment (0o-de-lally) [reply language/diem-framework/modules/0L/TowerState.move]: This wasn't quite ready for review. First getting the old tests to pass with new APIs, then change the params.

File: language/diem-framework/modules/0L/Globals.move [hunk 2]
```diff
@@ -111,8 +117,8 @@ module Globals {
           // uses "scaled representation", since there are no decimals.
           subsidy_ceiling_gas: 8640000 * coin_scale, // subsidy amount assumes 24 hour epoch lengths. Also needs to be adjusted for coin_scale the onchain representation of human readable value.
           vdf_difficulty: 5000000, // FYI approx 10 mins per proof on 2020 macbook pro 2.5 ghz quadcore
-          epoch_mining_thres_lower: 20,
-          epoch_mining_thres_upper: 240, // upper bound enforced at 6 mins per proof.
+          epoch_mining_thres_lower: 7, // NOTE: bootstrapping, allowance for operator error.
+          epoch_mining_thres_upper: 72, // upper bound enforced at 20 mins per proof.
```
Comment (sm86) [inline language/diem-framework/modules/0L/Globals.move:121]: I thought we wanted to change this to 52. Upper bound enforced at 27 mins right

Comment (0o-de-lally) [reply language/diem-framework/modules/0L/Globals.move:121]: I think we need a wider range at the start to see how much variation we get.

**Review by sm86 — APPROVED** [review]

**Pull request merged**